### ISSUE TITLE
ValueError: Operands could not be broadcast together with shapes (56, 56, 32) (112, 112, 32)

### ISSUE THREAD

**Issue by NevilleMthw** [keras-team/keras#16925]
**System information**.
- Have I written custom code (as opposed to using a stock example script provided in Keras): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Linux Ubuntu
- TensorFlow installed from (source or binary): source
- TensorFlow version (use command below): 2.4.1
- Python version: 3.8
- Bazel version (if compiling from source):
- GPU model and memory: Nvidia GeForce RTX 2080

**Describe the problem**.

I am trying to build a model with keras using the Functional API. However, I get an error relating to the ```Add()``` function of my residual blocks.

**Describe the current behavior**.
The current error I get is: ```ValueError: Operands could not be broadcast together with shapes (56, 56, 32) (112, 112, 32)```

**Standalone code to reproduce the issue**.
```
visible = Input(shape=(224, 224, 3))
x = Conv2D(32, (3, 3), padding='same', name='conv_1')(visible)
x = LeakyReLU(alpha=0.1)(x)
x = RandomFlip(mode='horizontal', seed=None)(x)
x = BatchNormalization()(x)
block_1 = MaxPool2D(pool_size=(2, 2), padding='same')(x)

x = Conv2D(32, (3, 3), padding='same', name='conv_2')(block_1)
x = LeakyReLU(alpha=0.1)(x)
x = BatchNormalization()(x)
x = MaxPool2D(pool_size=(2, 2), padding='same')(x)
block_2 = Add()([x, block_1])

x = Conv2D(64, (3, 3), padding='same', name='conv_3')(block_2)
x = LeakyReLU(alpha=0.1)(x)
x = BatchNormalization()(x)
x = MaxPool2D(pool_size=(2, 2), padding='same')(x)
block_3 = Add([x, block_2])

x = Conv2D(64, (3, 3), padding='same', name='conv_4')(block_3)
x = LeakyReLU(alpha=0.1)(x)
x = BatchNormalization()(x)
x = MaxPool2D(pool_size=(2, 2), padding='same')(x)
block_4 = Add([x, block_3])

x = Conv2D(64, (3, 3), padding='same', name='conv_5')(block_4)
x = LeakyReLU(alpha=0.1)(x)
x = BatchNormalization()(x)
x = MaxPool2D(pool_size=(2, 2), padding='same')(x)
block_5 = Add([x, block_4])

x = Conv2D(64, (3, 3), padding='same', name='conv_6')(block_5)
x = LeakyReLU(alpha=0.1)(x)
x = BatchNormalization()(x)
x = MaxPool2D(pool_size=(2, 2), padding='same')(x)
block_6 = Add([x, block_5])

x = Conv2D(128, (3, 3), padding='same', name='conv_7')(block_6)
x = LeakyReLU(alpha=0.1)(x)
x = BatchNormalization()(x)
x = MaxPool2D(pool_size=(2, 2), padding='same')(x)
block_7 = Add([x, block_6])

x = Conv2D(128, (3, 3), padding='same', name='conv_8')(block_7)
x = LeakyReLU(alpha=0.1)(x)
x = BatchNormalization()(x)
x = MaxPool2D(pool_size=(2, 2), padding='same')(x)
block_8 = Add([x, block_7])

x = Conv2D(128, (3, 3), padding='same', name='conv_9')(block_8)
x = LeakyReLU(alpha=0.1)(x)
x = BatchNormalization()(x)
x = MaxPool2D(pool_size=(2, 2), padding='same')(x)
block_9 = Add([x, block_8])

x = Conv2D(128, (3, 3), padding='same', name='conv_10')(block_9)
x = LeakyReLU(alpha=0.1)(x)
x = BatchNormalization()(x)
x = MaxPool2D(pool_size=(2, 2), padding='same')(x)
block_10 = Add([x, block_9])

x = Conv2D(256, (3, 3), padding='same', name='conv_11')(block_10)
x = LeakyReLU(alpha=0.1)(x)
x = BatchNormalization()(x)
x = MaxPool2D(pool_size=(2, 2), padding='same')(x)
block_11 = Add([x, block_10])

x = Conv2D(256, (3, 3), padding='same', name='conv_12')(block_11)
x = LeakyReLU(alpha=0.1)(x)
x = BatchNormalization()(x)
x = MaxPool2D(pool_size=(2, 2), padding='same')(x)
block_12 = Add([x, block_11])

x = Conv2D(256, (3, 3), padding='same', name='conv_13')(block_12)
x = LeakyReLU(alpha=0.1)(x)
x = BatchNormalization()(x)
x = MaxPool2D(pool_size=(2, 2), padding='same')(x)
block_13 = Add([x, block_12])

flat_layer = Flatten()(block_13)
x = Dropout(0.35)(flat_layer)
x = Dense(512, kernel_regularizer=regularizers.l2(0.1))(x)
x = LeakyReLU(alpha=0.1)(x)
x = Dropout(0.35)(x)

outputs = Dense(4)(x)

model = Model(visible, outputs, name='Functional_Model')

model.summary()
```

**Source code / logs**.

```
Output exceeds the size limit. Open the full output data in a text editor
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
/home/lunet/conm/Desktop/Stenosis-Project/functional.ipynb Cell 2 in <cell line: 12>()
     10 x = BatchNormalization()(x)
     11 x = MaxPool2D(pool_size=(2, 2), padding='same')(x)
---> 12 block_2 = Add()([x, block_1])
     14 x = Conv2D(64, (3, 3), padding='same', name='conv_3')(block_2)
     15 x = LeakyReLU(alpha=0.1)(x)

File ~/.conda/envs/stenosis/lib/python3.8/site-packages/tensorflow/python/keras/engine/base_layer.py:951, in Layer.__call__(self, *args, **kwargs)
    945 # Functional Model construction mode is invoked when `Layer`s are called on
    946 # symbolic `KerasTensor`s, i.e.:
    947 # >> inputs = tf.keras.Input(10)
    948 # >> outputs = MyLayer()(inputs)  # Functional construction mode.
    949 # >> model = tf.keras.Model(inputs, outputs)
    950 if _in_functional_construction_mode(self, inputs, args, kwargs, input_list):
--> 951   return self._functional_construction_call(inputs, args, kwargs,
    952                                             input_list)
    954 # Maintains info about the `Layer.call` stack.
    955 call_context = base_layer_utils.call_context()

File ~/.conda/envs/stenosis/lib/python3.8/site-packages/tensorflow/python/keras/engine/base_layer.py:1090, in Layer._functional_construction_call(self, inputs, args, kwargs, input_list)
   1086 if keras_tensor.keras_tensors_enabled():
   1087   with call_context.enter(
   1088       layer=self, inputs=inputs, build_graph=True, training=training_value):
   1089     # Check input assumptions set after layer building, e.g. input shape.
-> 1090     outputs = self._keras_tensor_symbolic_call(
   1091         inputs, input_masks, args, kwargs)
   1093     if outputs is None:
   1094       raise ValueError('A layer\'s `call` method should return a '
   1095                        'Tensor or a list of Tensors, not None '
   1096                        '(layer: ' + self.name + ').')

File ~/.conda/envs/stenosis/lib/python3.8/site-packages/tensorflow/python/keras/engine/base_layer.py:822, in Layer._keras_tensor_symbolic_call(self, inputs, input_masks, args, kwargs)
    820   return nest.map_structure(keras_tensor.KerasTensor, output_signature)
    821 else:
--> 822   return self._infer_output_signature(inputs, args, kwargs, input_masks)

File ~/.conda/envs/stenosis/lib/python3.8/site-packages/tensorflow/python/keras/engine/base_layer.py:862, in Layer._infer_output_signature(self, inputs, args, kwargs, input_masks)
    856 with backend.name_scope(self._name_scope()):
    857   with autocast_variable.enable_auto_cast_variables(
    858       self._compute_dtype_object):
    859     # Build layer if applicable (if the `build` method has been
    860     # overridden).
    861     # TODO(kaftan): do we maybe_build here, or have we already done it?
--> 862     self._maybe_build(inputs)
    863     outputs = call_fn(inputs, *args, **kwargs)
    865   self._handle_activity_regularization(inputs, outputs)

File ~/.conda/envs/stenosis/lib/python3.8/site-packages/tensorflow/python/keras/engine/base_layer.py:2710, in Layer._maybe_build(self, inputs)
   2705 if not hasattr(self.build, '_is_default'):
   2706   # Any setup work performed only once should happen in an `init_scope`
   2707   # to avoid creating symbolic Tensors that will later pollute any eager
   2708   # operations.
   2709   with tf_utils.maybe_init_scope(self):
-> 2710     self.build(input_shapes)  # pylint:disable=not-callable
   2711 # We must set also ensure that the layer is marked as built, and the build
   2712 # shape is stored since user defined build functions may not be calling
   2713 # `super.build()`
   2714 Layer.build(self, input_shapes)

File ~/.conda/envs/stenosis/lib/python3.8/site-packages/tensorflow/python/keras/utils/tf_utils.py:272, in shape_type_conversion.<locals>.wrapper(instance, input_shape)
    270 if input_shape is not None:
    271   input_shape = convert_shapes(input_shape, to_tuples=True)
--> 272 output_shape = fn(instance, input_shape)
    273 # Return shapes from `fn` as TensorShapes.
    274 if output_shape is not None:

File ~/.conda/envs/stenosis/lib/python3.8/site-packages/tensorflow/python/keras/layers/merge.py:112, in _Merge.build(self, input_shape)
    110   else:
    111     shape = input_shape[i][1:]
--> 112   output_shape = self._compute_elemwise_op_output_shape(output_shape, shape)
    113 # If the inputs have different ranks, we have to reshape them
    114 # to make them broadcastable.
    115 if None not in input_shape and len(set(map(len, input_shape))) == 1:
...
     85           'together with shapes ' + str(shape1) + ' ' + str(shape2))
     86     output_shape.append(i)
     87 return tuple(output_shape)

ValueError: Operands could not be broadcast together with shapes (56, 56, 32) (112, 112, 32)
```

**Comment by NevilleMthw** [comment]
I solved this problem by removing the maxPool layers as this was making my data shape smaller each time.

**Comment by sushreebarsa** [comment]
@NevilleMthw Thank you for the update!
